packages/frontend/src/components/profile/full_access_key_rotations/FullAccessKeyRotation.js

```diff
@@ -572,20 +572,29 @@ const FullAccessKeyRotation = ({ fullAccessKey }) => {
                         <span>
                             <Translate id='fullAccessKeys.transaction' />
                         </span>
-                        <Link>
-                            <a
-                                href={`${CONFIG.EXPLORER_URL}/txns/${transactionHash}`}
-                                target='_blank'
-                                rel='noreferrer'
-                            >
-                                {transactionHash}
-                            </a>
-                        </Link>
-                    </div>
-                    <div className='text-gray-500 text-sm mt-1'>
-                        <Translate id='fullAccessKeys.createdAt' />
-                        &nbsp;{createdAt.toLocaleString()}
+                        <div className='text-gray-500 text-sm'>
+                            <Translate id='fullAccessKeys.createdAt' />
+                            &nbsp;{createdAt.toLocaleString()}
+                        </div>
                     </div>
+                    <Link className='mt-1'>
+                        <a
+                            href={`${CONFIG.EXPLORER_URL}/txns/${transactionHash}`}
+                            target='_blank'
+                            rel='noreferrer'
+                        >
+                            {transactionHash.length > 15
+                                ? transactionHash
+                                      .substring(0, 6)
+                                      .concat(
+                                          '...',
+                                          transactionHash.substring(
+                                              transactionHash.length - 6
+                                          )
+                                      )
+                                : transactionHash}
```

Comment:
@SteveKok this code broke the /profile page while logged in. Please take a look ASAP.

Reply:
@Patrick1904 thansk for informing us.

This problem is caused by sometimes indexer didn't give us the keyMeta and transaction hash when a new key is created.

We have solve this problem in this PR:
https://github.com/mynearwallet/my-near-wallet/pull/201/files